### ISSUE TITLE
Add preview mode

### ISSUE THREAD

**Issue by pepijn-vanvlaanderen** [strapi/nextjs-corporate-starter#18]
Would be interesting to have a preview/draft mode included in this starter as well.

**Comment by abrorAbdullaev** [comment]
Even if we implement the preview feature in UI, can we extend the admin and add there a preview button somewhere without modifying the base module of @strapi/strapi?

**Comment by pepijn-vanvlaanderen** [comment]
> Even if we implement the preview feature in UI, can we extend the admin and add there a preview button somewhere without modifying the base module of @strapi/strapi?

Yes, there are already numerous Strapi plugins that do that, this one seems closest:
- https://market.strapi.io/plugins/strapi-plugin-preview-button

**Comment by PaulBratslavsky** [comment]
That is a great idea.  Will add it to the list.

**Comment by rrubio** [comment]
@pepijn-vanvlaanderen  were you successfull adding the preview buttom to this demo? i'm the getting some erorrs at my end. 

```
Unhandled Runtime Error
Error: Cannot read properties of undefined (reading 'attributes')

Source
src/app/[lang]/[...slug]/page.tsx (18:22) @ attributes

  16 | const page = await getPageBySlug(params.slug, params.lang);
  17 | 
> 18 | if (!page.data[0].attributes?.seo) return FALLBACK_SEO;
     |                  ^
  19 | const metadata = page.data[0].attributes.seo
  20 | 
  21 | return {
```

**Comment by pepijn-vanvlaanderen** [comment]
@rrubio it is actually very straightforward when you follow the guide here: https://nextjs.org/docs/app/building-your-application/configuring/draft-mode

You just need a custom /api/preview/route.ts where the draftMode gets enabled, after that cookie is set al your pages are dynamically rendered and you have your preview.

**Comment by PaulBratslavsky** [comment]
I am going to keep this issue open for anyone wanting to add this as a PR.  In terms of items on my list, this has lower priority.😅
